jwql/instrument_monitors/nircam_monitors/claw_monitor.py

```diff
@@ -0,0 +1,394 @@
+                             (abs(1 - (df_orig['mean'] / df_orig['median'])) < 0.05)]
+
+                # Plot the background levels over time
+                ax = fig.add_subplot(grid[i])
+                ax.scatter(df['expstart_mjd'], df['median'])
+
+                # Match scaling in all plots to the first detector with data. Shade median+/-10% region.
+                if len(df) > 0:
+                    if found_limits is False:
+                        first_med = np.nanmedian(df['median'])
+                        found_limits = True
+                    ax.set_ylim(first_med - first_med * 0.5, first_med + first_med * 0.5)
+                    med = np.nanmedian(df['median'])
+                    ax.axhline(med, ls='-', color='black')
+                    ax.axhspan(med - med * 0.1, med + med * 0.1, color='gray', alpha=0.4, lw=0)
+
+                # Axis formatting
+                ax.set_title(det, fontsize=40)
+                ax.set_xlim(start_mjd, end_mjd)
+                ax.set_xticks(time_tick_vals)
+                ax.set_xticklabels(time_tick_labels, fontsize=20, rotation=45)
+                ax.yaxis.set_tick_params(labelsize=20)
+                ax.set_ylabel('Background [MJy/sr]', fontsize=30)
+                # ax.set_xlabel('Date [YYYY-MM-DD]')
+                ax.grid(ls='--', color='gray')
+            fig.savefig(os.path.join(self.output_dir_bkg, '{}_backgrounds.png'.format(fltr)), dpi=180, bbox_inches='tight')
+            fig.clf()
+            plt.close()
+
+    def process(self):
+        """The main method for processing.  See module docstrings for further details.
+        """
+
+        # Get detector order and plot settings, depending on the wavelength channel
+        if self.wv == 'SW':
+            detectors_to_run = ['NRCA2', 'NRCA4', 'NRCB3', 'NRCB1', 'NRCA1', 'NRCA3', 'NRCB4', 'NRCB2']  # in on-sky order, don't change order
+            cols, rows = 5, 2
+            grid = plt.GridSpec(rows, cols, hspace=.2, wspace=.2, width_ratios=[1, 1, 1, 1, .1])
+            fig = plt.figure(figsize=(40, 20))
+            cbar_fs = 20
+            fs = 30
+        else:
+            detectors_to_run = ['NRCALONG', 'NRCBLONG']
+            cols, rows = 3, 1
+            grid = plt.GridSpec(rows, cols, hspace=.2, wspace=.2, width_ratios=[1, 1, .1])
+            fig = plt.figure(figsize=(20, 10))
+            cbar_fs = 10
+            fs = 20
+
+        # Make source-masked, median-stack of each detector's images
+        found_scale = False
+        for i, det in enumerate(detectors_to_run):
+            logging.info('Working on {}'.format(det))
+            files = self.files[self.detectors == det]
+            # Remove missing files; to avoid memory/speed issues, only use the first 20 files,
+            # which should be plenty to see any claws.
+            files = [f for f in files if os.path.exists(f)][0:20]
+            stack = np.ma.ones((len(files), 2048, 2048))
+            for n, f in enumerate(files):
```

Comment:
Same here with single character variable names if possible. `f` --> `fname`. I think `n` is fine here because of the python looping structure/iterator naming scheme.